src/components/inline_edit/inline_edit_form.tsx

```diff
@@ -27,53 +32,58 @@ import { useEuiI18n } from '../i18n';
-   * When the flag is true, the value will be saved. When the flag is false, the user will be
-   * returned to editMode.
-   */
-  onConfirm?: () => boolean;
-  /**
-   * Form label that appears above the form control
-   * This is required for accessibility because there is no visual label on the input
-   */
-  inputAriaLabel: string;
-  /**
-   * Aria-label for save button in editMode
-   */
-  saveButtonAriaLabel?: string;
-  /**
-   * Aria-label for cancel button in editMode
-   */
-  cancelButtonAriaLabel?: string;
-  /**
-   * Start in editMode
-   */
-  startWithEditOpen?: boolean;
-  /**
-   * Props that will be applied directly to the EuiEmptyButton displayed in readMode
-   */
-  readModeProps?: Omit<EuiButtonEmptyPropsForButton, 'onClick'>;
-  /**
-   * Props that will be applied directly to the `EuiFormRow` and `EuiFieldText` input displayed in editMode
-   */
-  editModeProps?: {
-    formRowProps?: Partial<EuiFormRowProps>;
-    inputProps?: Partial<EuiFieldTextProps>;
-  };
-  /**
-   * Loading state when changes are saved in editMode
-   */
-  isLoading?: boolean;
+export type EuiInlineEditCommonProps = HTMLAttributes<HTMLDivElement> &
+  CommonProps & {
+    defaultValue: string;
+    /**
+     * Allow users to pass in a function that is called when the confirm button is clicked
+     * The function should return a boolean flag that will determine if the value will be saved.
+     * When the flag is true, the value will be saved. When the flag is false, the user will be
+     * returned to editMode.
+     */
+    onConfirm?: (editModeValue: string) => boolean;
+    /**
+     * Form label that appears above the form control
+     * This is required for accessibility because there is no visual label on the input
+     */
+    inputAriaLabel: string;
+    /**
+     * Aria-label for save button in editMode
+     */
+    saveButtonAriaLabel?: string;
+    /**
+     * Aria-label for cancel button in editMode
+     */
+    cancelButtonAriaLabel?: string;
+    /**
+     * Start in editMode
+     */
+    startWithEditOpen?: boolean;
+    /**
+     * Props that will be applied directly to the EuiEmptyButton displayed in readMode
+     */
+    readModeProps?: Omit<EuiButtonEmptyPropsForButton, 'onClick'>;
+    /**
+     * Props that will be applied directly to the `EuiFormRow` and `EuiFieldText` input displayed in editMode
+     */
+    editModeProps?: {
+      formRowProps?: Partial<EuiFormRowProps>;
+      inputProps?: Partial<EuiFieldTextProps>;
+    };
+    /**
+     * Loading state when changes are saved in editMode
+     */
+    isLoading?: boolean;
 
-  /**
-   * Validation for the form control used to edit text in editMode
-   */
-  isInvalid?: boolean;
-};
+    /**
+     * Validation for the form control used to edit text in editMode
+     */
+    isInvalid?: boolean;
+    /**
+     * Returns the value of the EuiFieldText when the editMode save button is clicked
```

Comment:
I think we should clarify that this does *not* fire if the save button is clicked but onConfirm returns false.
```suggestion
     * Callback that passes the updated value of the edited text when the save button is pressed,
     * and the `onConfirm` callback (if passed) returns true
```

Reply:
Also, let's move `onSave` further up to be above `onConfirm` type definition - this is really kind of one of the more important APIs of the component